go/libkb/upak.go

```diff
@@ -23,35 +23,45 @@ func BaseProofSet(u *keybase1.UserPlusAllKeys) *ProofSet {
 
 // checkKIDPGP checks that the user has the given PGP KID valid *now*. Note that it doesn't
 // check for revoked PGP keys, and it also does not check key expiration.
-func checkKIDPGP(u *keybase1.UserPlusAllKeys, kid keybase1.KID) (found bool) {
-	for _, key := range u.PGPKeys {
-		if key.KID.Equal(kid) {
+func checkKIDPGP(u *keybase1.UserPlusKeysV2AllIncarnations, kid keybase1.KID) (found bool) {
+	for _, key := range u.Current.PGPKeys {
+		if key.Base.Kid.Equal(kid) {
 			return true
 		}
 	}
 	return false
 }
 
-func checkKIDKeybase(u *keybase1.UserPlusAllKeys, kid keybase1.KID) (found bool, revokedAt *keybase1.KeybaseTime, deleted bool) {
-	for _, key := range u.Base.DeviceKeys {
-		if key.KID.Equal(kid) {
-			return true, nil, false
+func checkKIDKeybase(u *keybase1.UserPlusKeysV2AllIncarnations, kid keybase1.KID) (found bool, revokedAt *keybase1.KeybaseTime, deleted bool) {
+	exportRevokedAt := func(key keybase1.PublicKeyV2NaCl) *keybase1.KeybaseTime {
+		if key.Base.Revocation != nil {
+			// This is the inverse of the marshalling we do in rpc_exim.go.
+			return &keybase1.KeybaseTime{
+				Unix:  key.Base.Revocation.Time,
+				Chain: int(key.Base.Revocation.PrevMerkleRootSigned.Seqno),
```

Comment:
We don't cast the time here (that would usually be a mistake), but we do cast the Chain seqno. Hopefully that's ok?